components/server/ee/src/prebuilds/bitbucket-app.ts

```diff
@@ -89,15 +93,38 @@ export class BitbucketApp {
                 return undefined;
             }
 
-            console.log('Starting prebuild.', { contextURL })
-            // todo@alex: add branch and project args
-            const ws = await this.prebuildManager.startPrebuild({ span }, { user, contextURL, cloneURL: data.gitCloneUrl, commit: data.commitHash});
+            console.debug('Bitbucket push event: Starting prebuild.', { contextURL });
+
+            const projectOwner = await this.findProjectOwner(data.gitCloneUrl);
+            const ws = await this.prebuildManager.startPrebuild({ span }, {
+                user: projectOwner?.user || user,
+                project: projectOwner?.project,
+                branch: data.branchName,
+                contextURL,
+                cloneURL: data.gitCloneUrl,
+                commit: data.commitHash
+            });
             return ws;
         } finally {
             span.finish();
         }
     }
 
+    protected async findProjectOwner(cloneURL: string): Promise<{ user: User, project?: Project } | undefined> {
+        const project = await this.projectDB.findProjectByCloneUrl(cloneURL);
+        if (project) {
+            const owner = !!project.userId
+                ? { userId: project.userId }
+                : (await this.teamDB.findMembersByTeam(project.teamId || '')).filter(m => m.role === "owner")[0];
```

Comment:
> Why is it useful to pick a random owner from the team here, as opposed to just picking the user that started the prebuild as before?

Another explainer: have a user A create a project in a team and then let user A leave the team, which account to use to tag the prebuild?

Reply:
@jankeromnes, can we discuss this in a call?

Reply:
Hm. I tend to agree with Jan that we should not try to do sth intimate (use token) that is unexpected.
How does the BitBucket "installation flow" look like? E.g., can we store a userId there that we can use here? That way we'd at least maintain the expectation that who registered a repo will also execute the prebuilds (like for GitHub apps).

> Another explainer: have a user A create a project in a team and then let user A leave the team, which account to use to tag the prebuild?

This sounds like a very interesting question to me: What is `projectId.userId` useful for, anyway? :thinking:

Reply:
**Suggestion**: For the sake of this PR, let's separate the "who owns the Team" and "who owns the BitBucket 'installation'" discussions into (a) separate issue(s). And simply exit here if `project.userId` is not set. Would that work, @AlexTugarev ?

Reply:
The point is that we're storing the users token in the webhook registration, which is part of history when we had no project introduced yet. I believe it would be more robust to add the project id in the webhook registration and use any available token of any team owner (with access to that BB repo) when doing updates to the webook registration. This is something we concluded while discussing about the ownership of a project within a team, and what should happen when one leaves a team, e.g. should the project be removed? 

I'm also ok with trying to maintain the installer's link as far as possible and fallback to a team owner after the installer was removed from the team.

Reply:
@jankeromnes and @geropl, I changed to prefer the previous webhook installer, which should meet the expectation in most cases. I suggest to overhaul the semantics for all providers in a separate issues, because it seems that we did not cover all cases. That's something we can and should cover in unit tests for clear communications.

Reply:
Sounds good!

Reply:
@jankeromnes and @geropl, thanks once again for pushing this! after re-re-reading the flow and testing once more, I realized that's a bad decision to limit to project owners 🙄. the fallback mechanism should, if applied, try to find and team member to rescue the state of the project, if the original installer was removed from the team. so, priority is to maintain the webhook membership and then to fallback to any of the team member with the relevant provider connection.

I took the change and aligned the gitlab app code as well, because that was the blueprint here. 

let's continue the clean up of this in the effort to provide a state of the webhook installation.